games/main.tex

```diff
@@ -0,0 +1,1123 @@
+\longnewglossaryentry{finger pointing}{name=finger pointing}{"litigation scheme allowing upstream peers to shift blame to downstream peer"}
+\longnewglossaryentry{certified delivery}{name=certified delivery}{message delivery with signed receipt from recipient}
+\longnewglossaryentry{swap index}{name=swap index}{sequential number of waivers}
+\longnewglossaryentry{object stream}{name=object stream}{series of messages passed from one node to its peer}
+\longnewglossaryentry{global balance}{name=global balance}{the balance of the chequebook contract}
+\longnewglossaryentry{waiver}{name=waiver}{promise not to cash out an outstanding cheque}
+\longnewglossaryentry{epoch}{name=epoch}{periodic interval of settlement for soft channel deposits}
+\longnewglossaryentry{handover state}{name=handover state}{root hash of an outgoing data stream signed against downstream peer and time}
+\longnewglossaryentry{takeover state}{name=takeover state}{handover state signed against initial state by downstream peer}
+\longnewglossaryentry{provable data exchange}{name=provable data exchange}{peer to peer object streams allowing handover and takeover proofs}
+\newcommand\gloss[1]{\emph{\gls{#1}}}
+
+
+\title{Generalised swap swear and swindle games}
+% \subtitle{Scalable infrastructure for decentralised service economies}
+\author{Viktor Trón}
+\date{\today}
+\begin{document}
+
+
+\maketitle
+\begin{abstract}
+\end{abstract}
+
+\tableofcontents
+
+\section{Introduction}
+Public proof-of-resource blockchains implement decentralised consensus on the validity and ordering of
+transactions in a state machine. Clients run a virtual machine to calculate state transitions by calling
+functions of programs sitting on accounts triggered by transactions. Given the expressive power of
+these programs called smart contracts, the blockchain can
+govern rules of interaction, enforce agreements, verify facts.
+Economic incentive systems driven by transparent smart contracts can regulate data flow
+so that emergent properties of peer-to-peer services are beneficial to users.
+It can act as a decentralised application platform running services potentially
+disintermediating trusted third parties traditionally needed for most domains of human interaction.
+
+While Turing-complete blockchain is the final missing piece of the puzzle to bring the cypherpunk vision
+of decentralisation to its completion, scalability has been repeatedly brought up as the major bottleneck
+in their mass adoption to real-world problems. Transaction costs for micropayments,
+speed and volume limitations are likely problematic even for evolved blockchains
+implementing sharding and proof of stake consensus.
+
+Side chains and state channels have been proposed to remedy these shortcomings.
+While inspired by these approaches, in this paper we take a different approach.
```

Comment:
use different word for "approaches" to avoid repetition: "while inspired by these ideas"....or something like that